### ISSUE TITLE
Additional use cases

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-processes#231]
Several capabilities could be useful in other projects:

- [x] Time-series fitting (see #240)
- [ ] Segmentation
- [ ] Spectral unmixing
- [ ] Regression
- [ ] Unsupervised classification
- [ ] Supervised classification (see #295)
- [ ] Spectral sharpening (see #221)
- [x] Convolutions (apply_kernel and apply_neighborhood)
- [x] Morphologic operations (apply_kernel and apply_neighborhood)
- [x] Spatial filtering (filter_spatial / load_collection)

We could also include image processing algorithms that allow transforming EO data collections into temporally equidistant and consistent time series datasets. Note: These techniques shall include as a minimum compositing (e.g. best-pixel, weighted averaging, max value), statistical aggregation (e.g. percentile, mean, standard deviation) and might additionally include capabilities for deriving coefficients or statistics of time series fits (e.g. integral, amplitude, inflection points, etc.).

req. 44 + 45

**Comment by jdries** [comment]
convolutions == apply_kernel?
spectral_sharpening == resolution_merge (have a pull request for that!)
morhpological operations => dilation, erosion => apply_kernel

**Comment by jdries** [comment]
supervised classification => random forest => use case 9@SRR3
regression => use case 8 @SRR3

**Comment by jonathom** [comment]
**Spectral Unmixing**

I had a look and came up with two use cases that could be helpful towards developing processes. First, a **quick overview** of what I understand is contained in an unmixing process:
1. Preprocessing from spectral radiance to corrected reflectance values (important, but not part of the unmixing itself)
2. selecting or providing endmembers (spectral values of the materials or land covers in question)
3. unmixing, linear or non-linear, both with multiple possible approaches (finding the combination of endmembers that lead to the observed spectrum of pixels)
4. outputs are maps per endmember, describing the fraction of the pixels that is covered by the endmember

**UC1**: I want to know the fractions of materials/land cover that each pixel in my datacube contains. For this, I have a labeled dataset (geojson etc.) with endmember polygons that I confirmed as containing pure areas (material-wise). For preparation I need to aggregate mean values per endmember class from the satellite collection that I will be using. I want to use a standard linear unmixing method, in which the combined reflectances should sum to one. The result should be a datacube with as many bands as endmembers that I defined, with an optional time dimension remaining untouched.

**UC2**: I want to compare different unmixing processes on a certain dataset, e.g. different linear and non-linear approaches. For this, I already extracted the spectral values of my endmembers per band of the collection in question and would like to input them as an array. I would like to select different unmixing methods and receive the output as stated in UC1, together with a band containing the residual error that the unmixing process couldn't attribute to an endmember, per pixel.

GEE doesn't bother with different unmixing methods and [only offers one](https://developers.google.com/earth-engine/apidocs/ee-image-unmix), but it seems that different methods can have pros and cons. Unfortunately I do not know which methods are must-haves, so methods are excluded from these use cases.
